### ISSUE TITLE
kit_with_splits_2023.npz

### ISSUE THREAD

**Issue by Stefano-retinize** [rd20karim/M2T-Interpretable#1]
Hello.

I was interested in using this code for a dataset purpose but, can't find this file: C:\\Users\\karim\\PycharmProjects\\HumanML3D\\kit_with_splits_2023.npz'

Thanks for your help @rd20karim

**Comment by rd20karim** [comment]
Hi @Stefano-retinize,
You can find it by following the processing requirements mentioned in the [README](https://github.com/rd20karim/M2T-Interpretable?tab=readme-ov-file#preprocess-datasets), which will direct you to this [project link section](https://github.com/rd20karim/M2T-Segmentation?tab=readme-ov-file#preprocess-dataset)
then you can change the path to match your project path directory.

**Comment by Stefano-retinize** [comment]
Sorry maybe I am doing something wrong, but the .npz file you can download from (https://drive.google.com/drive/folders/1zkJ4LnGV64Ne5ZlRy2aWH1sXcH_aVFF8) doesn't seem to be the same. As an example in your code: [Line 21] ( https://github.com/rd20karim/M2T-Interpretable/blob/2d402d64007b581c21e496ee501571d2f3f71e23/datasets/kit_h3mld.py#L21)

but there is no key `["splits_ids"]` inside that .npy file either the key `['old_desc']`.

Please Appreciate your help
Thanks

**Comment by rd20karim** [comment]
@Stefano-retinize 

The link provided allows you to download the original KIT-ML version. In this work, I use an augmented version stored as `kit_with_splits_2023.npz.` To generate this file along with the corresponding corrected text annotations, please follow the steps [here](https://github.com/rd20karim/M2T-Segmentation?tab=readme-ov-file#preprocess-dataset).

You can also refer to my detailed answer [here](https://github.com/rd20karim/M2T-Segmentation/issues/2#issuecomment-2091363950).

For this project, you will only need the augmented  version.

Feel free to ask if you have questions!

**Comment by Stefano-retinize** [comment]
Hey @rd20karim  thanks for your prompts answers. I think I am missing something, in the [Link](https://drive.google.com/drive/folders/1D3bf2G2o4Hv-Ale26YW18r1Wrh7oIAwK) , there is no kit_with_splits_2023.npz file.

Thanks again!

**Comment by rd20karim** [comment]
@Stefano-retinize 

If you want to generate text based on your custom dataset (e.g., Mocap data), it needs to be converted to match the scale and format of either KIT-ML (21 joints) or HumanML3D (22 joints). The key` 'all_motions'` provides access to all motion samples, where each sample is a sequence of 3D coordinates (xyz) representing human skeleton joints. If you don’t have target text available, you can modify the code to generate predictions without performing quantitative evaluation.

The dataset loader scripts in the` datasets` folder should be updated accordingly.

You'll need to download the previous datasets to construct the vocabulary object. I recommend reviewing the entire discussion [here](https://github.com/rd20karim/M2T-Segmentation/issues/2#issuecomment-2091935280).

The link is for downloading the raw KIT-ML data, which will need to be processed.

**Comment by Stefano-retinize** [comment]
Oh If I am doing things right now, I think my confusion was because there is no build_dataset.py in this repo. And I couldn't see where it was telling me to run that in the readme actually. I go the .npz now.
I also had this error: ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (6016,) + inhomogeneous part
which I solved doing: kit_poses = np.array(kit_poses, dtype=object)

**Comment by rd20karim** [comment]
Yes, some of these errors and their solutions are mentioned in this discussion [link](https://github.com/rd20karim/M2T-Segmentation/issues/2#issuecomment-2091935280)
The steps mentioned in the other project should use their own files.
